cypress/global.d.ts

```diff
@@ -17,8 +17,10 @@ limitations under the License.
 import "matrix-js-sdk/src/@types/global";
```

Comment:
> Please request review once you get the code passing CI.

 - ❌ [cypress: 1-register/register.spec.ts (merge)](https://dashboard.cypress.io/projects/ppvnzg/runs/797/test-results?actions=%5B%5D&browsers=%5B%5D&groups=%5B%5D&isFlaky=%5B%5D&modificationDateRange=%7B%22startDate%22%3A%221970-01-01%22%2C%22endDate%22%3A%222038-01-19%22%7D&orderBy=EXECUTION_ORDER&oses=%5B%5D&specs=%5B%5D&statuses=%5B%7B%22value%22%3A%22FAILED%22%2C%22label%22%3A%22FAILED%22%7D%5D&testingTypesEnum=%5B%5D&utm_source=github)
    - This one feels flakey? It works locally.
 - ❌ [Static Analysis / Typescript Syntax Check (pull_request)](https://github.com/matrix-org/matrix-react-sdk/runs/6620089042?check_suite_focus=true)
     - For the `Property 'setValue' does not exist on type 'SettingsStore'`, I'm looking for help on how to solve that one: https://github.com/matrix-org/matrix-react-sdk/pull/8354#discussion_r878660677
     - For the rest, you mentioned [`You're importing something which uses jsx (or imports something that does) , the cypress tsconfig isn't configured for jsx.`](https://matrix.to/#/!bEWtlqtDwCLFIAKAcv:matrix.org/$XdTxFttFojekJWT3W8TtkM19sI_M6t66S35WxlGgIkU?via=matrix.org&via=element.io&via=mozilla.org). It seems like all of this stems from `global.d.ts` and the `import type SettingsStore from "../src/settings/SettingsStore";` import there. I guess that relates to the first point here. Any tips?

Reply:
We can probably just enable jsx support?

Reply:
> We can probably just enable jsx support?

Sounds reasonable 👍 Can I get some assistance from the Web team to add this? That seems a bit orthogonal to this PR.

Reply:
@t3chguy It doesn't seem to be that simple. I've split out what I have so far in https://github.com/matrix-org/matrix-react-sdk/pull/8738 (help wanted)